docs/automated_testing.md

```diff
@@ -0,0 +1,509 @@
+
+```
+Feature: Interviews load
+
+  Scenario: The 209A loads
+    Given I start the interview at "ma_209a_package"
+
+  Scenario: The Plaintiff's Motion to Modify loads
+    Given I start the interview at "plaintiffs_motion_to_modify_209a"
+```
+
+More complex tests might wait till your code is pretty much how you want it. Every time you change your variable names, you may have to update the tests.
+
+## Story tables syntax
+
+Very basically, a **story table** contains a list of variables and values. The order in which you list the rows does not matter, which gives a lot more flexibility when developers are changing around the order of questions. Whenever pages with those variables appear, they will be given the value listed.
+
+Tip: it can be easy to lose track of what variables you've included, so it can be useful keep the rows in alphabetical order.
+
+You can have multiple tables per Scenario and you can put other steps between tables. Remember that each table needs to state the `id` of the `question` where it should stop.
+
+### Story table description
+
+The step that triggers a story table is
+
+```
+    And I get to the question id "some id!" with this data:
+```
+
+It needs to know the `id` of the page where it will stop. You can find the `id` in the `question` block in the YAML (or using the Sources tab in a live interview). Under that, you put the header row of the table:
+
+```
+      | var | value | trigger |
+```
+
+This labels the three columns you will use. Under that, you add a row for every field that you want to interact with during the interview. Start with a blank row then fill it in:
+```
+      |  |  |  |
+```
+
+### var
+
+In the **var** column, write the name of the variable you want to set **as it appears on question's screen**. Most times you can see that value in the YAML `question` block. Sometimes it will be generated by code in the `question` block. If you're not sure what name some code generates, you may have to ask the folks who wrote the code.
+
+<!-- The Document Assembly Line library has built in questions that use such code.
+- `users[i].name_fields()` generates all the name fields with the right index number (NOT 'i'). E.g. `users[0].name.first`, `users[0].name.suffix`, `users[1].name.first`, `users[1].name.suffix`
+- `users[i].address_fields()` generates all the address fields with the right index number (NOT 'i'). E.g. `users[0].address.address`, `users[1].address.address` -->
+
+### value
+
+In the **value** column, write what you want the field to be set to. For checkboxes, `true` means 'checked' and `false` means 'unchecked'.
+
+
+### trigger
+
+In the **trigger** column, write the name of the intrinsic name of the variable that will trigger the page for this field. This column is optional in most cases. In cases where the page is using an [index variable](https://docassemble.org/docs/fields.html#index%20variables) or a [`generic object`](https://docassemble.org/docs/modifiers.html#generic%20object).
```

Comment:
```suggestion
In the **trigger** column, write the name of the intrinsic name of the variable that will trigger the page for this field. This column is necessary in cases where the page is using an [index variable](https://docassemble.org/docs/fields.html#index%20variables) or a [`generic object`](https://docassemble.org/docs/modifiers.html#generic%20object), and optional everywhere else.
```

Reply:
I like it and I'll add one improvement where I see what I think is a typo of mine and another clarification:

```suggestion
In the **trigger** column, write the intrinsic name of the variable that will trigger the page where this field appears. This column is necessary in cases where the page is using an [index variable](https://docassemble.org/docs/fields.html#index%20variables) or a [`generic object`](https://docassemble.org/docs/modifiers.html#generic%20object), and optional everywhere else.
```